ra2ce/graph/vector_network_wrapper.py

```diff
@@ -0,0 +1,370 @@
+        )  # TODO make consistant convention with osm
+        if not nodes.crs:
+            nodes.crs = self.crs
+        if not edges.crs:
+            edges.crs = self.crs
+
+        if include_nodes is True:
+            return edges, nodes
+        else:
+            return edges
+
+    def setup_network_from_vector(
+        self,
+    ) -> Tuple[nx.MultiGraph, gpd.GeoDataFrame]:
+        """
+        Sets up a network from vector files.
+
+        Returns
+        -------
+        nx.MultiGraph
+            MultiGraph representing the graph.
+        gpd.GeoDataFrame
+            GeoDataFrame representing the network.
+        """
+        file = self.network_dict["file"]
+        file_crs = self.network_dict["file_crs"]
+        is_directed = self.network_dict["is_directed"]
+
+        gdf = self.read_vector(fn=file, crs=file_crs)
+        gdf = self.clean_vector(
+            gdf, explode_and_deduplicate_geometries=True
+        )  # TODO move explode_and_deduplicate_geometries to [project] or [cleanup]
+        g = self.setup_graph_from_vector(gdf, is_directed=is_directed)
+        edges, nodes = self.setup_network_from_graph(g, include_nodes=True)
+        graph_complex = nut.graph_from_gdf(edges, nodes, node_id="node_fid")
+        return graph_complex, edges
+
+    def read_vector(
+        self, fn: Union[str, Path, List[Union[str, Path]]], crs: Union[int, str]
+    ) -> gpd.GeoDataFrame:
+        """
+        Reads a vector file or a list of vector files.
+        Clips for project region and reproject to project crs if available.
+
+        Parameters
+        ----------
+        fn : Union[str, Path, List[Union[str, Path]]]
+            Path to the vector file or list of paths.
+        crs : Union[int, str]
+            Coordinate reference system for the file. Allow only one crs for all fn.
+
+        Returns
+        -------
+        gpd.GeoDataFrame
+            GeoDataFrame representing the vector data.
+        """
+        gdf = self._read_file(fn)
+        if gdf is None:
+            logger.info("no file is read.")
+            return None
+
+        # set crs and reproject if needed
+        if gdf.crs is None and crs is not None:
+            gdf = gdf.set_crs(CRS.from_user_input(crs))
+            logger.info("setting crs as default EPSG:4326. specify crs if incorrect")
+
+        if self.crs is not None:
+            gdf = gdf.to_crs(self.crs)
+            logger.info("reproject vector file to project crs")
+
+        # clip for region
+        if self.region is not None:
+            gdf = gpd.overlay(gdf, self.region, how="intersection", keep_geom_type=True)
+            logger.info("clip vector file to project region")
+
+        # validate
+        if len(gdf) == 0:
+            logger.warning("No vector features found within project region")
+            return None
+
+        return gdf
+
+    def _read_file(
+        self, fn: Union[str, Path, List[Union[str, Path]]]
```

Comment:
force the usage of Path, we can't support dual types over the whole solution.

```suggestion
        self, file_list: list[Path]
```